File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        length = len(nums)

        l, r, ans = [0] * length, [0] * length, [0] * length

        l[0] = 1
        for i in range(1, length):
            l[i] = l[i-1] * nums[i-1]
        
        r[length-1] = 1
        for i in range(length-2, -1, -1):
            r[i] = r[i+1] * nums[i+1]
        

        for i in range(length):
            ans[i] = l[i] * r[i]
        
        return ans

File: Data Structures & Algorithms/two-integer-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let map = new Map()

        let i=0
        while(i < nums.length){
            let curr = nums[i]
                let diff = target-curr
              let  diffI = map.get(diff)
              
            if (diffI != undefined) {
                return [diffI, i]
            }
            map.set(curr, i)

            i++
        }
    }
}
